src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -783,6 +790,15 @@ internal bool IsAutoProperty
                 return null;
             }
         }
+
+        private bool AllowsInitializer
+        {
+            get
+            {
+                // PROTOTYPE(semi-auto-props): Implement this.
+                return IsAutoProperty;
```

Comment:
Will this require extra binding? 😕

Reply:
It will require _some_ amount of binding, certainly, but I think it won't require fully binding the body. We'll just need to check to see whether any identifiers named `field` are used without a `field` being in scope.

Reply:
> We'll just need to check to see whether any identifiers named `field` are used without a `field` being in scope.

Any pointer on how can I see if there are identifiers named `field` without needing to fully bind the bind? Introducing a new `BinderFlag` for example? (I know it's unlikely we want to introduce a binder flag for this, but can't think of the correct way to do it)

Reply:
My thought is that we can walk the binder chain, looking for anything in scope named `field`. IE, locals or fields. I do think we should ask the LDM what this code means though:

```cs
int Prop
{
    set
    {
        field = value;
        {
            int field = 0; // Does this shadow the backing field? Or is there an error reported somewhere?
        }
    }
}
```

Reply:
@333fred Do I need to open an issue for that in dotnet/csharplang?

Reply:
I think there is no design question here. The specification is clear: "The `field` identifier is only considered the `field` keyword when there is  no existing symbol named `field` in scope at that location." We can certainly have different implementation strategies, but for now I would stick with the simplistic approach of simply binding the body. Anything else is more complexity and more risk.

Reply:
> @333fred Do I need to open an issue for that in dotnet/csharplang?

Yes, let's get that in an issue, or in the spec.